server/svix-server/src/worker.rs

```diff
@@ -193,20 +193,24 @@ async fn dispatch(
                 response_status_code: Set(status_code),
                 response: Set(body),
                 status: Set(status),
-
                 ..attempt
             };
-            Ok(attempt)
+            match status {
+                MessageStatus::Success => Ok(attempt),
+                _ => Err((
+                    attempt,
+                    crate::error::Error::Generic(format!("HTTP STATUS {}", status_code)),
```

Comment:
Generic feels wrong. Does it actually need to be a `crate::error::Error`? Why not just a string error?